org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/correction/LocalCorrectionsSubProcessor.java

```diff
@@ -1353,65 +1355,24 @@ public static void createNewTypeAsPermittedSubTypeProposal(IInvocationContext co
 		proposals.add(new NewCUUsingWizardProposal(compilationUnit, null, NewCUUsingWizardProposal.K_CLASS, sealedTypeElement,  typeBinding, relevance + 6, false));
 	}
 
-	public static void addTypeAsPermittedSubTypeProposal(IInvocationContext context, IProblemLocation problem, Collection<ICommandAccess> proposals) throws JavaModelException {
-		ASTNode selectedNode= problem.getCoveringNode(context.getASTRoot());
-		if (selectedNode == null) {
-			return;
-		}
-		if (!ASTHelper.isSealedTypeSupportedInAST(selectedNode.getAST())) {
-			return;
-		}
+	public static void addTypeAsPermittedSubTypeProposal(IInvocationContext context, IProblemLocation problem, Collection<ICommandAccess> proposals) {
+		SealedClassFixCore fix = SealedClassFixCore.addTypeAsPermittedSubTypeProposal(context.getASTRoot(), (IProblemLocationCore)problem);
+		if (fix != null) {
```

Comment:
In the future, when you want to do more refactoring like this where you move the code into a CompilationUnitRewriteOperationsFixCore, the following code could/should be put into a utility method where you pass it the fix, the image or image id, ICompilationUnit, whatever, and you get back a CUCorrectionProposal to add.

Reply:
I could probably fix this to get it right. I actually intended to add another 1 or 2 quick-fixes into `SealedClassFixCore`.

Reply:
Actually, I may keep this as-is for now and maybe we can adjust in a future PRs. If we provide some utility that does that I think it's just a matter of ensuring future refactorings use it.